kit/scrolls/src/Lists/UIMasonryList.tsx

```diff
@@ -335,6 +339,28 @@ function useVirtualization<Item>(
         [initialWindowHeight, onLayoutProp, setWidth],
     );
 
+    const maybeCallOnEndReached = React.useCallback(
+        (contentLength: number, visibleLength: number) => {
+            const distanceFromEnd = contentLength - visibleLength;
+            const threshold =
+                onEndReachedThreshold != null ? onEndReachedThreshold * contentLength : 2;
+            if (
+                onEndReached &&
+                distanceFromEnd < threshold &&
+                contentLength !== sentEndForContentLength.current
+            ) {
+                // Only call onEndReached once for a given content length
+                sentEndForContentLength.current = contentLength;
+                onEndReached({ distanceFromEnd });
+            } else if (distanceFromEnd > threshold) {
+                // If the user scrolls away from the end and back again cause
+                // an onEndReached to be triggered again
+                sentEndForContentLength.current = 0;
+            }
```

Comment:
Let's avoid unwanted reassignment
```suggestion
            } else if (sentEndForContentLength.current != 0 && distanceFromEnd > threshold) {
                // If the user scrolls away from the end and then returns back again,
                // it should cause the `onEndReached` callback to be triggered again
                sentEndForContentLength.current = 0;
            }
```